### PR TITLE
Refactor/react class components to react functional components

### PR DESCRIPTION
**Pull request by dawid-pstrak-rmtcfm-com**

### CODE REVIEW

File: frontend/src/components/AddTask.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
+import { Button, TextField, Typography } from '@material-ui/core';
+
+const AddItem = props => {
+    const [name, setName] = useState('');
+
+    const { handleAddTask } = props;
```
Comment (ewelina-szczotka-prpl) [inline frontend/src/components/AddTask.js]: you can destructure props above, so `const AddItem = ({handleAddTask}) => ...`, same in other components

File: frontend/src/components/TaskList.js [hunk 2]
```diff
@@ -1,153 +1,120 @@
...
 
-            this.setState({
-                fetching: false
-            });
-        } catch (error) {
-            console.error(error);
-            this.setState({ isLoggedIn: false });
-        }
-    };
+        setIsFetching(false);
```
Comment (ewelina-szczotka-prpl) [inline frontend/src/components/TaskList.js]: does the above action actually wait for tasks being fetched or is it immediately changing `isFetching`  to false? 

you should have `isFetching` or `isLoading` variable in reducer in store for tasks and get it here from store

File: frontend/src/components/TaskList.js [hunk 3]
```diff
@@ -1,153 +1,120 @@
...
-
-        const notificationsComponents = notifications.map(notification => (
-            <Notification
-                key={notification.ts}
-                variant={notification.variant}
-                message={notification.message}
-                onClose={this.closeNotification}
-                ts={notification.ts}
+    const tasksList =
+        !isFetching &&
```
Comment (ewelina-szczotka-prpl) [inline frontend/src/components/TaskList.js]: if `isFetching` is true you could display some loader

File: frontend/src/components/AddTask.js (hunk 1 above)
Comment (dawid-pstrak-rmtcfm-com) [reply frontend/src/components/AddTask.js]: done

File: frontend/src/components/TaskList.js (hunk 2 above)
Comment (dawid-pstrak-rmtcfm-com) [reply frontend/src/components/TaskList.js]: It was working, but refactored it

File: frontend/src/components/TaskList.js (hunk 3 above)
Comment (dawid-pstrak-rmtcfm-com) [reply frontend/src/components/TaskList.js]: done

**Review by ewelina-szczotka-prpl — APPROVED** [review]

**Pull request merged**